estimagic/differentiation/richardson_extrapolation.py

```diff
@@ -0,0 +1,275 @@
+            that ``step_ratio`` >= 1.
+
+        exponentiation_step (int): Step representing the growth of the exponent in
+            the series expansions of the limit.
+            For central difference derivative approximation ``exponentiation_step`` = 2.
+
+        order (int): Initial order of the approximation error of sequence elements.
+            For central difference derivative approximation ``order`` = 2.
+
+    Returns:
+        coef (np.ndarray): Richardson coefficients, array has length num_terms + 1.
+
+    Example:
+    >>> import numpy as np
+    >>> num_terms = 2
+    >>> step_ratio = 2.
+    >>> exponentiation_step = 2
+    >>> order = 2
+    >>> _richardson_coefficients(num_terms, step_ratio, exponentiation_step, order)
+    array([ 0.02222222, -0.44444444,  1.42222222])
+
+    """
+    rows, cols = np.ogrid[: num_terms + 1, :num_terms]
+
+    coef_mat = np.ones((num_terms + 1, num_terms + 1))
+    coef_mat[:, 1:] = (1.0 / step_ratio) ** (
+        rows @ (exponentiation_step * cols + order)
+    )
+
+    coef = pinv(coef_mat)[0]
+    return coef
+
+
+def _estimate_error(new_seq, old_seq, richardson_coef):
+    """Estimate error of multiple Richardson limit approximation.
+
+    Args:
+        new_seq (np.ndarray): Multiple estimates of the limit of ``old_seq``. The last
+            two dimensions coincide with those of ``old_seq``. The first dimensions
+            denotes the number of different estimates.
+
+        old_seq (np.ndarray): The sequence of which we want to approximate the limit.
+            Has dimension (k x n x m), where k denotes the number of sequence elements
+            and an element ``sequence[l, :, :]`` denotes the (n x m) dimensional element
+
+        richardson_coef (np.ndarray): Richardson coefficients. See function
+            ``_richardson_coefficient`` for details.
+
+    Returns:
+        abserr (np.ndarray): The error estimate for each limit approximation in
+            ``new_seq``.
+
+    """
+    new_seq_len = new_seq.shape[0]
+
+    unnormalized_covariance = np.sum(richardson_coef ** 2)
+    fact = np.maximum(TQUANTILE * np.sqrt(unnormalized_covariance), EPS * 10.0)
+
+    if new_seq_len <= 1:
+        delta = np.diff(old_seq, axis=0)
+        tol = np.maximum(np.abs(old_seq[:-1]), np.abs(old_seq[1:])) * fact
+        err = np.abs(delta)
+        converged = err <= tol
+        abserr = err[-new_seq_len:] + np.where(
+            converged[-new_seq_len:],
+            tol[-new_seq_len:] * 10,
+            abs(new_seq - old_seq[-new_seq_len:]) * fact,
+        )
+    else:
+        err = np.abs(np.diff(new_seq, axis=0)) * fact
+        tol = np.maximum(np.abs(new_seq[1:]), np.abs(new_seq[:-1])) * EPS * fact
+        converged = err <= tol
+        abserr = err + np.where(
+            converged, tol * 10, abs(new_seq[:-1] - old_seq[-new_seq_len + 1 :]) * fact,
+        )
+
+    return abserr
+
+
+def _get_order_and_exponentiation_step(method):
```

Comment:
I think this is my favorite function in your PR! It just takes makes something that I could not understand after hours of reading numdifftools and makes it really simple to understand. Only thing that could be improved: Explain what order and exponentiation step are in vey first paragraph. Currently you explain it in the Returns section, which makes sense, but is a bit late.